### ISSUE TITLE
GSL bug in lsqfit.empbayes_fit for integer z0

### ISSUE THREAD

**Issue by millernb** [gplepage/lsqfit#5]
Consider the slight modification of `z0` in the empirical Bayes example (the second-to-last line). 

```python
import numpy as np
import gvar as gv
import lsqfit

# fit data
x = np.array([1., 2., 3., 4.])
y = np.array([3.4422, 1.2929, 0.4798, 0.1725])

# prior
prior = gv.gvar(['10(1)', '1.0(1)'])

# fit function
def fcn(x, p):
    return p[0] * gv.exp( - p[1] * x)

# find optimal dy
def fitargs(z):
    dy = y * z
    newy = gv.gvar(y, dy)
    return dict(data=(x, newy), fcn=fcn, prior=prior)

fit, z = lsqfit.empbayes_fit(1, fitargs, step=0.1)
print(fit.format(True))
```

(I've also tweaked `step` to aid convergence).

When GSL is used and z0 is an integer (or at least, not explicitly a float), the call will fail with error:
```python
TypeError: unsupported operand type(s) for *: 'float' and 'NoneType'
```

This error can be traced back to [this line](https://github.com/gplepage/lsqfit/blob/22c0a2ade8b627ab3da4b8a380b4612dfa69e9e1/src/lsqfit/_gsl.pyx#L107), I think, in `_gsl.pyx`. Essentially, `array2vector` expects the first argument to be an numpy array of `dtype=float`, but instead we have `dtype=int`, so the call fails. Notably, the above example works properly when scipy is used as the minimization backend instead.

The error can be fixed by forcing `z0buff` inside the `empbayes_fit` function to be a float.
```python
z0buff = np.array(z0buff, dtype=float)
```

**Comment by gplepage** [comment]
Fixed in v11.8.

**Comment by gplepage** [comment]
Thanks!
